spotipy/client.py

```diff
@@ -1562,26 +1562,15 @@ def _get_uri(self, type, id):
         return "spotify:" + type + ":" + self._get_id(type, id)
 
     def _search_multiple_markets(self, q, limit, offset, type, markets, total):
-        results = {
-                type + 's': {
-                    'href': [],
-                    'items': [],
-                    'limit': limit,
-                    'next': None,
-                    'offset': 0,
-                    'previous': None,
-                    'total': 0
-                }
-            }
+        results = {}
+        types = type + 's'
```

Comment:
Just realised that `type` can be a comma separated list of types such as `album,track`

https://developer.spotify.com/documentation/web-api/reference/search/search/

So the type would need to be split and looped over. I checked, if we do `search(limit=4, `track,artist`), it will return 4 tracks and 4 artists.

---------------

Another possibility is to "fix" `total` at the beginning of the method so that if it exists, it is not less than limit. If `limit` > `total`, e.g. `search_markets(limit=4, total=2, markets=['US', 'GB', 'AU])`, then show a warning that says something like:
> Warning: `limit` was automatically adjusted to equal `total` as `total` must equal or be higher than `limit`.

Reply:
I was unaware of type having the ability to be multiple values. I can edit this too in addition to making a new `search_market` method

Reply:
Thanks a lot @NLeRoy917 🙏

Reply:
No problem! Just to make sure I digested it all properly:
* Revert `search` to its original method
* Create a new `search_markets` method incorporating the changes
* Check for multiple types and loop through to incorporate all the results
* Auto-adjust limit to total value if total is less than limit

Reply:
All good apart from number 3 that I think you shouldn't need anymore if you already do number 4

Reply:
Right - I believe this means that total will be the total returned for each country instead of total for ALL countries.